sql-infra: add resource control feature (#12353)
pingcap/docs: tikv-configuration-file.md

+ The maximum number of TSOs in a timestamp request.
+ In a default TSO physical time update interval (`50ms`), PD provides at most 262144 TSOs. When requested TSOs exceed this number, PD provides no more TSOs. This configuration item is used to avoid exhausting TSOs and the reverse impact of TSO exhaustion on other businesses. If you increase the value of this configuration item to improve high availability, you need to decrease the value of [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) at the same time to get enough TSOs.
+ Default value: `8192`

> ## resource_control

Reviewer: ```suggestion
## resource-control
```

> Configuration items related to resource control related to the TiKV storage layer.

Reviewer 2: ```suggestion
Configuration items related to resource control of the TiKV storage layer.
```

### `enabled` <span class="version-mark">New in v6.6.0</span>

> + Whether to support scheduling for user foreground read/write requests according to Request Unit (RU) of the corresponding resource group. For information about TiDB resource groups and resource control, see [TiDB resource control](/tidb-resource-control.md).

Reviewer 3: Shall we also link Request Unit (RU) to its definition page

> + Whether to support scheduling for user foreground read/write requests according to Request Unit (RU) of the corresponding resource group. For information about TiDB resource groups and resource control, see [TiDB resource control](/tidb-resource-control.md).

Author: > Shall we also link Request Unit (RU) to its definition page

Done

> + It is only meaningful to enable this configuration item if [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB. When it is enabled, TiKV will use the priority queue to schedule the queued requests for user foreground read/write requests. The priority of scheduling is inversely related to the amount of resources already consumed by the resource group where the request is located, and positively related to the quota of the corresponding resource group.

Reviewer 2: ```suggestion
+ Enabling this configuration item only works when [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB. When this configuration item is enabled, TiKV will use the priority queue to schedule the queued read/write requests from foreground users. The scheduling priority of a request is inversely related to the amount of resources already consumed by the resource group that receives this request, and positively related to the quota of the corresponding resource group.
```

> + Whether to support prioritized scheduling for user foreground read/write requests according to the corresponding resource group quota. For information about TiDB resource groups and resource control, see [TiDB resource control](/tidb-resource-control.md)
> + It is only meaningful to enable this configuration item if [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB side. When it is enabled, TiKV will use the priority queue to schedule the queued requests for user foreground read/write requests. The priority of scheduling is inversely related to the amount of resources already consumed by the resource group where the request is located, and positively related to the quota of the corresponding resource group.
> + Default value: `false`, which means to disable scheduling based on resource group quotas.

Author: ```suggestion
+ Whether to support scheduling for user foreground read/write requests according to Request Unit (RU) of the corresponding resource group. For information about TiDB resource groups and resource control, see [TiDB resource control](/tidb-resource-control.md).
+ It is only meaningful to enable this configuration item if [`tidb_enable_resource_control](/system-variables.md#tidb_enable_resource_control-new-in-v660) is enabled on TiDB. When it is enabled, TiKV will use the priority queue to schedule the queued requests for user foreground read/write requests. The priority of scheduling is inversely related to the amount of resources already consumed by the resource group where the request is located, and positively related to the quota of the corresponding resource group.
+ Default value: `false`, which means to disable scheduling based on the RU of the resource group.
```
